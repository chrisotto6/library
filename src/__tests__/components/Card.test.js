import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../../components/card'

const book = {
  title: 'Test (The Testing #1)',
  title_without_series: 'Test',
  authors: {
    author: {
      name: 'Chris Otto',
    },
  },
  image_url:
    'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1589998653l/6043781._SX98_.jpg',
  link: 'https://www.goodreads.com/book/show/6043781-blood-of-elves',
}

const rating = ['4']

test('App Snapshot Matches', () => {
  const tree = renderer.create(<Card book={book} rating={rating} />).toJSON()
  expect(tree).toMatchSnapshot()
})
