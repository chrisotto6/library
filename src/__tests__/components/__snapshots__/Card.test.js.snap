// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Matches 1`] = `
<li
  className="bookCard"
>
  <header>
    <div>
      <img
        alt="Test (The Testing #1)"
        src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1589998653l/6043781._SX98_.jpg"
      />
    </div>
  </header>
  <section
    className="cardContainer"
  >
    <h4
      className="h4 small"
    >
      Test
    </h4>
    <h5
      className="h5 small"
    >
      Chris Otto
    </h5>
    <div
      className="dv-star-rating dv-star-rating-non-editable"
      style={
        Object {
          "display": "inline-block",
          "position": "relative",
        }
      }
    >
      <input
        checked={false}
        className="dv-star-rating-input"
        id="_5"
        name=""
        onChange={[Function]}
        style={
          Object {
            "display": "none",
            "marginLeft": -9999,
            "position": "absolute",
          }
        }
        type="radio"
        value={5}
      />
      <label
        className="dv-star-rating-star dv-star-rating-empty-star"
        htmlFor="_5"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#333",
            "cursor": "default",
            "float": "right",
          }
        }
      >
        <i
          style={
            Object {
              "fontStyle": "normal",
            }
          }
        >
          ★
        </i>
      </label>
      <input
        checked={true}
        className="dv-star-rating-input"
        id="_4"
        name=""
        onChange={[Function]}
        style={
          Object {
            "display": "none",
            "marginLeft": -9999,
            "position": "absolute",
          }
        }
        type="radio"
        value={4}
      />
      <label
        className="dv-star-rating-star dv-star-rating-full-star"
        htmlFor="_4"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#ffb400",
            "cursor": "default",
            "float": "right",
          }
        }
      >
        <i
          style={
            Object {
              "fontStyle": "normal",
            }
          }
        >
          ★
        </i>
      </label>
      <input
        checked={false}
        className="dv-star-rating-input"
        id="_3"
        name=""
        onChange={[Function]}
        style={
          Object {
            "display": "none",
            "marginLeft": -9999,
            "position": "absolute",
          }
        }
        type="radio"
        value={3}
      />
      <label
        className="dv-star-rating-star dv-star-rating-full-star"
        htmlFor="_3"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#ffb400",
            "cursor": "default",
            "float": "right",
          }
        }
      >
        <i
          style={
            Object {
              "fontStyle": "normal",
            }
          }
        >
          ★
        </i>
      </label>
      <input
        checked={false}
        className="dv-star-rating-input"
        id="_2"
        name=""
        onChange={[Function]}
        style={
          Object {
            "display": "none",
            "marginLeft": -9999,
            "position": "absolute",
          }
        }
        type="radio"
        value={2}
      />
      <label
        className="dv-star-rating-star dv-star-rating-full-star"
        htmlFor="_2"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#ffb400",
            "cursor": "default",
            "float": "right",
          }
        }
      >
        <i
          style={
            Object {
              "fontStyle": "normal",
            }
          }
        >
          ★
        </i>
      </label>
      <input
        checked={false}
        className="dv-star-rating-input"
        id="_1"
        name=""
        onChange={[Function]}
        style={
          Object {
            "display": "none",
            "marginLeft": -9999,
            "position": "absolute",
          }
        }
        type="radio"
        value={1}
      />
      <label
        className="dv-star-rating-star dv-star-rating-full-star"
        htmlFor="_1"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "color": "#ffb400",
            "cursor": "default",
            "float": "right",
          }
        }
      >
        <i
          style={
            Object {
              "fontStyle": "normal",
            }
          }
        >
          ★
        </i>
      </label>
    </div>
    <div
      className="link"
    >
      <a
        className="h5 small"
        href="https://www.goodreads.com/book/show/6043781-blood-of-elves"
        rel="noopener noreferrer"
        target="_blank"
      >
        Goodreads Page
      </a>
    </div>
  </section>
</li>
`;
